tests/pipeline/test_engine.py

```diff
@@ -1424,23 +1429,22 @@ def test_regression_of_returns_factor(self,
                             expected_regression_results[i]
 
             for output in outputs:
-                assert_frame_equal(
-                    output_results[output],
-                    DataFrame(
-                        expected_output_results[output],
-                        index=self.dates[start_date_index:end_date_index + 1],
-                        columns=assets,
-                    ),
+                output_result = output_results[output]
+                expected_output_result = DataFrame(
+                    where(expected_mask, expected_output_results[output], nan),
+                    index=dates[start_date_index:end_date_index + 1],
+                    columns=assets,
                 )
+                assert_frame_equal(output_result, expected_output_result)
 
     def test_correlation_and_regression_with_bad_asset(self):
         """
         Test that `RollingPearsonOfReturns`, `RollingSpearmanOfReturns` and
         `RollingLinearRegressionOfReturns` raise the proper exception when
         given a nonexistent target asset.
         """
-        start_date_index = 6
-        end_date_index = 10
+        start_date_index = 14
```

Comment:
I changed this so that I could run the pipeline over a contiguous set of 5 days (i.e. with no weekend) while also having plenty of days to look back on. It was previously starting at index 6 only because that was the first day that gave enough days to look back on (specifically, you need at least as many days as the window_length of `returns` plus the correlation/regression length). I made this change above as well for consistency and it also simplified things when computing the expected masks to have a contiguous 5 days.